### PR TITLE
Convert morgan to be an object instead of a global function

### PR DESCRIPTION
**Pull request by ficristo**
This is a prerequisite for https://github.com/expressjs/morgan/pull/75.
In this PR I removed the deprecated code too, so I think this need a major update.
**--edit**
I changed the API too.
Before: `morgan('dev')`
After: `morgan().getLogger('dev')`

### CODE REVIEW

**Comment by dougwilson** [comment]
Please change the API back to the Node.js standard of a factory export.

**Comment by dougwilson** [comment]
The README needs to be updated for all the changes you made as well.

File: index.js [hunk 1]
```diff
@@ -97,8 +84,51 @@ function morgan(format, options) {
...
+ * @param {string} name
+ * @param {function} fn
+ * @public
+ */
+Morgan.token = function (name, fn) {
+  gTokens[name] = fn
+}
+
+Morgan.prototype.getLogger = function (format) {
+  assert (
```
Comment (dougwilson) [inline index.js]: This needs to throw `TypeError`s, not `AssertionError`s

File: index.js [hunk 2]
```diff
@@ -97,8 +84,51 @@ function morgan(format, options) {
...
+    format,
+    'Morgan.getLogger(format): format is mandatory'
+  );
+  assert (
```
Comment (dougwilson) [inline index.js]: This needs to throw `TypeError`s, not `AssertionError`s

**Comment by ficristo** [comment]
> Please change the API back to the Node.js standard of a factory export.

Could you clarify, maybe with an example?

**Comment by dougwilson** [comment]
> > Please change the API back to the Node.js standard of a factory export.

> Could you clarify, maybe with an example?

`app.use(morgan('dev'))`

**Comment by ficristo** [comment]
I submitted a new PR with a different method in https://github.com/expressjs/morgan/pull/90

**Pull request closed without merging**